src/Core/Configurations/JsonConfigSchemaValidator.cs

```diff
@@ -87,8 +76,15 @@ public async Task<JsonSchemaValidationResult> ValidateJsonConfigWithSchemaAsync(
             {
                 // Send a GET request to the URL specified in runtimeConfig.Schema to get the JSON schema.
                 HttpResponseMessage response = await _httpClient.GetAsync(runtimeConfig.Schema);
-                string jsonSchema = await response.Content.ReadAsStringAsync();
-                return jsonSchema;
+                if (response.IsSuccessStatusCode)
+                {
+                    string jsonSchema = await response.Content.ReadAsStringAsync();
+                    return jsonSchema;
+                }
+                else
+                {
+                    _logger!.LogError($"Failed to get schema from url: {runtimeConfig.Schema}");
+                }
```

Comment:
nit: not blocking this PR, but this error is no different than the error printed in the `catch` block. The `catch` block is catching specific errors like no connect, bad uri, timeouts, but this else statement catches an http response not equal to 2XX. We can give user more informative message to get this working.

Also, any errors sourced from failure to resolve schema file from URI should be logged as warnings because we have a fallback method of reading the schema file from the local repo. Any failure from the fallback method should definitely still be logged as "error"

Reply:
makes sense. thanks!